feat: allow setting autoscaling options to deployed KService (#374)
boson-project/func: docs/guides/func_yaml.md

  path: /workspace/configmap
```

### `options`

> Options allows you to set specific configuration for the deployed funciton, ie. it allows you to tweak Knative Service options related to autoscaling and other properties, if these options are not set in the default ones will be used. This options are based on the ones provided by `kn service` [command](https://github.com/knative/client/blob/main/docs/cmd/kn_service_create.md#options).

Reviewer: ```suggestion
Options allows you to set specific configuration for the deployed funciton, allowing you to tweak Knative Service options related to autoscaling and other properties. If these options are not set, the Knative defaults will be used. 
```

Minor wordsmithing, and I think it might be best to avoid referencing the `kn` options, just to avoid confusion among users - but I'm not firm on this suggestion.

Reviewer: Oh also a typo... `funciton`.

- `scale-min`: Minimum number of replicas.
- `scale-max`: Maximum number of replicas.
- `scale-init`: Initial number of replicas with which a Service starts. Can be 0 or a positive integer.
- `autoscale-window`: Duration to look back for making auto-scaling decisions. The Service is scaled to zero if no request was received in during that time. (eg: 10s)
- `concurrency-limit`: Hard Limit of concurrent requests to be processed by a single replica.
- `concurrency-target`: Recommendation for when to scale up based on the concurrent number of incoming request. Defaults to `concurrency-limit` when given.

> - `concurrency-utilization`: Percentage of concurrent requests utilization before scaling up.

Reviewer: It might be beneficial to have the default values listed as well.

Author: I was thinking about documenting default values, but those are based on a entries in a config in knative-serving namespace, so they could vary across individual installations.
